CASSANDRA-17259: January 2022 blog post for Apache Cassandra Changelog #11 (#89)
apache/cassandra-website: site-content/source/modules/ROOT/pages/blog/Apache-Cassandra-Changelog-11-January-2022.adoc

= Apache Cassandra Changelog #11
:page-layout: single-post
:page-role: blog-post
:page-post-date: January 18, 2022
:page-post-author: The Apache Cassandra Community
:description: The Apache Cassandra Community
:keywords: 

image::blog/changelog_header.jpg[Apache Cassandra Changelog]
Our monthly roundup of key activities and knowledge to keep the community informed.

== Release Notes

=== Released

> Available since 7th September, the latest release of Apache Cassandra is  https://www.apache.org/dyn/closer.lua/cassandra/4.0.1[4.0.1,window=_blank], (https://archive.apache.org/dist/cassandra/4.0.1/apache-cassandra-4.0.1-bin.tar.gz.asc[pgp^],  https://archive.apache.org/dist/cassandra/4.0.1/apache-cassandra-4.0.1-bin.tar.gz.sha256[sha256^], and https://archive.apache.org/dist/cassandra/4.0.1/apache-cassandra-4.0.1-bin.tar.gz.sha512[ sha512^]). This was a rapid release to fix a https://issues.apache.org/jira/browse/CASSANDRA-16877[critical bug^] in Gossip on large clusters. Please read the https://gitbox.apache.org/repos/asf?p=cassandra.git;a=blob_plain;f=CHANGES.txt;hb=refs/tags/cassandra-4.0.1[release notes^] and let us know if you encounter any problems.

Reviewer: I'd like to propose that we link to github instead of gitbox so new users know where to find the release notes. WDYT? 🙂

```suggestion
Available since 7th September, the latest release of Apache Cassandra is  https://www.apache.org/dyn/closer.lua/cassandra/4.0.1[4.0.1,window=_blank], (https://archive.apache.org/dist/cassandra/4.0.1/apache-cassandra-4.0.1-bin.tar.gz.asc[pgp^],  https://archive.apache.org/dist/cassandra/4.0.1/apache-cassandra-4.0.1-bin.tar.gz.sha256[sha256^], and https://archive.apache.org/dist/cassandra/4.0.1/apache-cassandra-4.0.1-bin.tar.gz.sha512[ sha512^]). This was a rapid release to fix a https://issues.apache.org/jira/browse/CASSANDRA-16877[critical bug^] in Gossip on large clusters. Please read the https://github.com/apache/cassandra/blob/cassandra-4.0.1/CHANGES.txt[release notes^] and let us know if you encounter any problems.
```

Note: As the docs are not yet updated, the bintray location for Debian users is replaced with the https://apache.jfrog.io/artifactory/cassandra/[ASF's JFrog Artifactory location^].

See the https://cassandra.apache.org/download/[download section^] for the latest stable and older supported versions of source and binary distributions.

To stay up-to-date, we recommend joining the Cassandra  https://cassandra.apache.org/community/[mailing list^].

== Community Notes

_Updates on Cassandra Enhancement Proposals (CEPs), how to contribute, and other community activities._ 

> _Are you new to the project? We have established a https://issues.apache.org/jira/secure/RapidBoard.jspa?rapidView=484[New Release tracking Kanboard^] and a https://issues.apache.org/jira/secure/RapidBoard.jspa?rapidView=484&quickFilter=2162&quickFilter=2160[“Starter Tickets”^] quick label that corresponds to our Low Hanging Fruit status. Any of these tickets should be of appropriate complexity for someone new to the project to tackle. You can also reach out in the #cassandra-dev Slack channel on the-asf.slack.com and ask for assistance from any of the 13 designated Cassandra mentors!

Reviewer: There's an underscore (`_`) at the start of this paragraph which looks out-of-place. Is the first sentence supposed to be in italics?

Otherwise:

```suggestion
Are you new to the project? We have established a https://issues.apache.org/jira/secure/RapidBoard.jspa?rapidView=484[New Release tracking Kanboard^] and a https://issues.apache.org/jira/secure/RapidBoard.jspa?rapidView=484&quickFilter=2162&quickFilter=2160[“Starter Tickets”^] quick label that corresponds to our Low Hanging Fruit status. Any of these tickets should be of appropriate complexity for someone new to the project to tackle. You can also reach out in the #cassandra-dev Slack channel on the-asf.slack.com and ask for assistance from any of the 13 designated Cassandra mentors!
```

> _Are you new to the project? We have established a https://issues.apache.org/jira/secure/RapidBoard.jspa?rapidView=484[New Release tracking Kanboard^] and a https://issues.apache.org/jira/secure/RapidBoard.jspa?rapidView=484&quickFilter=2162&quickFilter=2160[“Starter Tickets”^] quick label that corresponds to our Low Hanging Fruit status. Any of these tickets should be of appropriate complexity for someone new to the project to tackle. You can also reach out in the #cassandra-dev Slack channel on the-asf.slack.com and ask for assistance from any of the 13 designated Cassandra mentors!

Reviewer: I'd like to suggest adding a link to https://cassandra.apache.org/_/community.html so new users can find out how to get access to ASF Slack. 👍

Read PMC member Josh McKenzie’s https://lists.apache.org/thread/9rzwl3s2fr734sdc5jynbxz3z4vht3df[latest bi-weekly update^] for ongoing discussions and the latest on ticket progress.

===  Advisory

Apache Cassandra is _not_ vulnerable to the log4shell vulnerability, as you can see in Jira ticket, https://issues.apache.org/jira/browse/CASSANDRA-5883[CASSANDRA-5883^], Apache Cassandra never chose to use log4j2 preferring logback instead.

=== Announced

> During December we ran an https://issues.apache.org/jira/issues/?jql=project%20%3D%20CASSANDRA%20AND%20labels%20%3D%20AdventCalendar2021%20[Advent Calendar (Twitter initiative^] of new tickets to get started on. This was a great way for new project contributors to pick up hand-curated tickets to dive into.

Reviewer: Missing closing parenthesis:

```suggestion
During December we ran an https://issues.apache.org/jira/issues/?jql=project%20%3D%20CASSANDRA%20AND%20labels%20%3D%20AdventCalendar2021%20[Advent Calendar (Twitter initiative)^] of new tickets to get started on. This was a great way for new project contributors to pick up hand-curated tickets to dive into.
```

image::blog/advent-calendar.png[Apache Cassandra’s advent calendar campaign on Twitter]

=== Announced

It may be cold outside, but we’re warming ourselves up with thoughts of a great Google Summer of Code, and there are some exciting changes this year! Read https://lists.apache.org/thread/58v2bvfzwtfgqdx90qmm4tmyoqzsgtn4[Paulo Motta’s post^] on the mailing list. 

=== Announced

Ubuntu 21.10 includes Apache Cassandra https://www.devopsdigest.com/canonical-releases-ubuntu-2110[“packaged as a snap, enabling DataOps teams to set up a Cassandra cluster”^], and the latest LTS Docker Images now include Apache Cassandra v4.

=== Announced

Apple indicates support and encourages further contribution to the Apache Cassandra community with a https://opensource.apple.com/projects/[new online portal^] covering all its open source projects and contributions.

=== Added

The Project Management Committee (PMC) is pleased to announce that *Sumanth Pasupuleti* has been invited to https://lists.apache.org/thread/kvsvltom5p79bk2r1j52sxl3y3qjl71w[become a committer^], and he has accepted! Thank you for all your contributions Sumanth, and congratulations! 👏

=== Added

> An https://cwiki.apache.org/confluence/display/CASSANDRA/Roadmap[up-to-date roadmap^] for the project has been added to the wiki.

Reviewer: I'd like to suggest using the permanent URLs (permalink) for cwiki pages so that when the wiki titles get changed, the permalink will continue to work.

```suggestion
An https://cwiki.apache.org/confluence/x/gYvkCw[up-to-date roadmap^] for the project has been added to the wiki.
```

=== Added

> Andrés de la Peña opened a discussion about https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-3%3A+Guardrails[CEP-3^], a guardrails feature that is intended to be an easy way to “enforce system-wide soft and hard limits to prevent anti-patterns of bad usage”. In the long-term this is intended to prevent the severe degradation of a node/cluster’s performance through user actions such as having too many secondary indexes, too large partitions, almost full disks, and so on. This feature passed a vote, has been developed and merged! Congrats Andrés!

Reviewer: Use cwiki permalink:

```suggestion
Andrés de la Peña opened a discussion about https://cwiki.apache.org/confluence/x/wxkRCQ[CEP-3^], a guardrails feature that is intended to be an easy way to “enforce system-wide soft and hard limits to prevent anti-patterns of bad usage”. In the long-term this is intended to prevent the severe degradation of a node/cluster’s performance through user actions such as having too many secondary indexes, too large partitions, almost full disks, and so on. This feature passed a vote, has been developed and merged! Congrats Andrés!
```

=== Added

> https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-10%3A+Cluster+and+Code+Simulations[CEP-10 Cluster and Code Simulation^] passed a vote and Benedict Elliot Smith has merged in the simulator changes so can expect to see some interesting things showing up regarding correctness on our system from the new testing paradigms this enables.

Reviewer: Use cwiki permalink:

```suggestion
https://cwiki.apache.org/confluence/x/LY7OCg[CEP-10 Cluster and Code Simulation^] passed a vote and Benedict Elliot Smith has merged in the simulator changes so can expect to see some interesting things showing up regarding correctness on our system from the new testing paradigms this enables.
```

=== Discussed

Josh McKenzie is seeking more feedback on the topic of keeping a releasable trunk and our quality on the project. You can join https://lists.apache.org/thread/8xt4tqb3w4j3jyxj859o3vs8f5xjgqg8[the discussion here^]

=== Passed

> After clarification that the API for distributed transactions will be modular / pluggable, CEP-15 General Purpose Transactions passed. This work will allow for experimentation in the future with other algorithms. As a community we're excited to see how the project evolves!

Reviewer: Add link to CEP-15:

```suggestion
After clarification that the API for distributed transactions will be modular / pluggable, https://cwiki.apache.org/confluence/x/FQRACw[CEP-15 General Purpose Transactions^] passed. This work will allow for experimentation in the future with other algorithms. As a community we're excited to see how the project evolves!
```

=== Passed

> https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-16%3A+Auth+Plugin+Support+for+CQLSH[CEP-16 Auth Plugin Support for CQLSH^], proposed by Stefan Miklosovic, will enable switching to a new enhanced authentication scheme while retaining the ability to use cqlsh in normal operation.  A vote was taken and passed and the feature is in development.

Reviewer: Use cwiki permalink:

```suggestion
https://cwiki.apache.org/confluence/x/8AtACw[CEP-16 Auth Plugin Support for CQLSH^], proposed by Stefan Miklosovic, will enable switching to a new enhanced authentication scheme while retaining the ability to use cqlsh in normal operation.  A vote was taken and passed and the feature is in development.
```

=== Passed

> https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-17%3A+SSTable+format+API[CEP-17^] by Jacek Lewandowski is around an SSTable format API. The proposal is for a design that enables for optionally plugging-in alternative implementations of SSTables and for testing the individual implementations in isolation. You can follow the https://lists.apache.org/thread.html/r636bebcab4e678dbee042285449193e8e75d3753200a1b404fcc7196%40%3Cdev.cassandra.apache.org%3E[discussion here^]. A vote was taken and passed and this CEP is in development.

Reviewer: Use cwiki permalink:

```suggestion
https://cwiki.apache.org/confluence/x/sotnCw[CEP-17^] by Jacek Lewandowski is around an SSTable format API. The proposal is for a design that enables for optionally plugging-in alternative implementations of SSTables and for testing the individual implementations in isolation. You can follow the https://lists.apache.org/thread.html/r636bebcab4e678dbee042285449193e8e75d3753200a1b404fcc7196%40%3Cdev.cassandra.apache.org%3E[discussion here^]. A vote was taken and passed and this CEP is in development.
```

=== Closed

> CEP-18 was under discussion to add more modularity to a few different subsystems. It was decided to close the CEP to enable ideas to be considered individually, rather than grouped under one feature proposal.

Reviewer: Add link to CEP-18:

```suggestion
https://cwiki.apache.org/confluence/x/5YtnCw[CEP-18^] was under discussion to add more modularity to a few different subsystems. It was decided to close the CEP to enable ideas to be considered individually, rather than grouped under one feature proposal.
```

== User Space

=== Witfoo

“Humankind is at a pivotal point in history, that requires us to work together to find new innovative ways to process the petabytes of data we create every day, and the WitFoo R&D team has found a friend in Cassandra through our endeavors to meet big-data needs in cybersecurity operations.” - https://www.witfoo.com/blog/our-move-from-elastic-to-cassandra/[Charles Herring^]

=== Backblaze

“In our experience, nothing beats Apache Cassandra for write scaling, and we’re looking forward to the performance and management improvements in the 4.0 release. We rely on Cassandra to help manage nearly two exabytes of customer data and serve over 500 billion files for our customers across 175 countries so optimizing Cassandra’s capabilities and performance means a lot to us.” - https://www.devprojournal.com/technology-trends/open-source/whats-included-in-the-cassandra-4-0-release-2/[Elliott Sims^]

Do you have a Cassandra case study to share? Email cassandra@constantia.io. 

== In the News

> Dev Pro: https://www.devprojournal.com/technology-trends/open-source/whats-included-in-the-cassandra-4-0-release/[What’s Included in the Cassandra 4.0 Release?^]

Reviewer: Insert a newline:

```suggestion
Dev Pro: https://www.devprojournal.com/technology-trends/open-source/whats-included-in-the-cassandra-4-0-release/[What’s Included in the Cassandra 4.0 Release?^]

```
